### ISSUE TITLE
Quarto render command renaming and revert renaming folders at every render

### ISSUE THREAD

**Issue by Stramon1um** [quarto-dev/quarto-cli#9774]
Hi,

It has been happening only recently, and I don't know what the reason behind it could be. Briefly: every time I execute the command quarto render, apart from rendering the .qmd files to .html, as expected, it also renames the folder docs/site_libs to docs/site_libs 2, breaking the layout of my website. Then, if I re-run quarto render, it reverts the names to docs/site_libs, and everything is fine. The screenshot shows the last execution of quarto render.

Did this happen to someone else before? Or do you know by chance what it is all about?

Thanks!

I'm running the last version of RStudio on macOS Sonoma. Quarto version 1.4.554.

<img width="1638" alt="2024-05-27_11 28 10" src="https://github.com/quarto-dev/quarto/assets/11859033/3c741877-c3e4-454f-93ad-3758bbde7372">

**Comment by Stramon1um** [comment]
Ok, I'm back with a small update.

I added a new blog post as planned, but it appeared again. I had to render twice to revert to normality, so it wasn't the extension after all.

I did some experiments and a lot of `quarto render` runs 😅 but I might have noticed something. Without adding or modifying any files or folders, I started rendering continuously until it appeared! 
It seems that when you render a .qmd file (which is actually a `quarto preview`), open the viewer and change page/s within the viewer like browsing a website (both RStudio and Safari), and then run `quarto render` again, it renames `docs/site_libs` to `docs/site_libs 2`. Whereas, when you render the .qmd page without navigating to any other page, it doesn't seem to appear.

I also remembered that the issue started when I updated quarto to 1.4.553. I created the whole website on quarto 1.3.450. Unfortunately, this is the only way I _managed_ to reproduce this weird behaviour. I don't know if some files are corrupted or if it more related to RStudio+quarto or else.

Thanks!

**Comment by mcanouil** [comment]
Unfortunately, this does help more. We really need something to work on such as a small plain text document, a Git repository.

Without, we are unable to provide you with any assistance.

**Comment by Stramon1um** [comment]
I see. Currently, I've been using Quarto only in one public repository, which is my personal website repository (see my profile).  As written above, I'm running Quarto v1.4.554 in RStudio v2024.04.1 Build 748, macOS Sonoma 14.5. 

I would like to give you more info or examples on how to replicate this issue, but I post not so often on my website, and most importantly, my skills are very basic in all of this. :)

**Comment by cderv** [comment]
We understand. 

It seems it does not happen always, and only in some specific situation. I suggest we keep this open and you can come back when you identified more situation where this happens. We can also see if someone else is noticing the same - for now, I never noticed this on Windows myself. 

You can also share your personal repo and we can try reproduce from there. 

Thanks for the report.

**Comment by Stramon1um** [comment]
Ok, I'm back again!

I might have found what was causing me this non-continuous issue. I'm still not 100% sure, but it seems to have disappeared once and for all. I believe it was the iCloud sync.

Using GitHub Desktop, it set the directory of all my repositories in my Documents by default, which iCloud also syncs. So far, so good. As I wrote earlier, I remembered that this "renaming issue" started when I started embedding PDF files in my webpage.  Today, I had the same issue rendering my last post. I tried to purge the `doc` and `.quarto` folders to re-render the whole project, and I found copies of the index.html, like `index 2.html`, randomly left in my post folder (which contains only the .qmd files. After another `quarto render`, I got new problems. At the same time, I noticed that iCloud was struggling to sync files*...Only at this point, I removed the whole repository from iCloud, cloned the last version outside of iCloud and ran `quarto render`...and then again, previewed some pages, moved within the previewed website and rendered again a couple of times more....nothing, no more renaming, no more issue of this kind!

Probably, when involving larger files, like PDF, the continuous sync of iCloud + the render/regeneration of folders/files does not work very well; I guess Quarto renders too fast to allow iCloud to sync/desync large files simultaneously, and maybe that's why I was having copies of folders and files from time to time....or at least this is my theory 😁

By the way, with the next post, I will be much more sure, although so far, it seems to be working flawlessly, no matter how many times I render.

**Comment by mcanouil** [comment]
To summarise, you are rendering in a cloud synced folder? If so, it's unfortunately not a big surprise that this does not work well.

**Comment by Stramon1um** [comment]
> To summarise, you are rendering in a cloud synced folder? If so, it's unfortunately not a surprise that this does not work well.

Yes, I was. But as I mentioned, everything was fine until I started involving PDFs. Now I know...
